packages/smooth_app/lib/helpers/background_task_helper.dart

```diff
@@ -0,0 +1,181 @@
+        switch (processName) {
+          case IMAGE_UPLOAD_TASK:
+            return uploadImage(inputData.data!, localDatabase);
+
+          case PRODUCT_EDIT_TASK:
+            return otherDetails(inputData.data!, localDatabase);
+
+          default:
+            return TaskResult.success;
+        }
+      },
+      listener: (Task task, TaskStatus status) {});
+  return TaskResult.success;
+}
+
+Future<TaskResult> otherDetails(
+    Map<String, dynamic> inputData, LocalDatabase localDatabase) async {
+  final BackgroundOtherDetailsInput inputTask =
+      BackgroundOtherDetailsInput.fromJson(inputData);
+  final Map<String, dynamic> mp =
+      json.decode(inputTask.inputMap) as Map<String, dynamic>;
+  final User user =
+      User.fromJson(jsonDecode(inputTask.user) as Map<String, dynamic>);
+  await OpenFoodAPIClient.saveProduct(
+    user,
+    Product.fromJson(mp),
+    language: LanguageHelper.fromJson(inputTask.languageCode),
+    country: CountryHelper.fromJson(inputTask.country),
+  );
+  final DaoProduct daoProduct = DaoProduct(localDatabase);
+  final ProductQueryConfiguration configuration = ProductQueryConfiguration(
+    inputTask.barcode,
+    fields: ProductQuery.fields,
+    language: LanguageHelper.fromJson(inputTask.languageCode),
+    country: CountryHelper.fromJson(inputTask.country),
+  );
+
+  final ProductResult queryResult =
+      await OpenFoodAPIClient.getProduct(configuration);
+  if (queryResult.status == 1) {
+    final Product? product = queryResult.product;
+    if (product != null) {
+      await daoProduct.put(product);
+      localDatabase.notifyListeners();
+    }
+  }
+  // Returns true to let platform know that the task is completed
+  return TaskResult.success;
+}
+
+Future<TaskResult> uploadImage(
+    Map<String, dynamic> inputData, LocalDatabase localDatabase) async {
+  final BackgroundImageInputData inputTask =
+      BackgroundImageInputData.fromJson(inputData);
+  final User user =
+      User.fromJson(jsonDecode(inputTask.user) as Map<String, dynamic>);
+  final SendImage image = SendImage(
+    lang: LanguageHelper.fromJson(inputTask.languageCode),
+    barcode: inputTask.barcode,
+    imageField: ImageFieldExtension.getType(inputTask.imageField),
+    imageUri: Uri.parse(inputTask.imageUri),
+  );
+  await OpenFoodAPIClient.addProductImage(user, image);
+  // go to the file system and delete the file that was uploaded
+  final File file = File(inputTask.imageUri);
+  file.deleteSync();
+  final DaoProduct daoProduct = DaoProduct(localDatabase);
+  final ProductQueryConfiguration configuration = ProductQueryConfiguration(
+    inputTask.barcode,
+    fields: ProductQuery.fields,
+    language: LanguageHelper.fromJson(inputTask.languageCode),
+    country: CountryHelper.fromJson(inputTask.country),
+  );
+
+  final ProductResult queryResult =
+      await OpenFoodAPIClient.getProduct(configuration);
+  if (queryResult.status == 1) {
```

Comment:
Please use a constant for "1" and explain what it means

Reply:
I found something similar in 
line 37 `lib\query\barcode_product_query.dart`
So just used the same block of code here

Reply:
It seems to be sent by the server https://github.com/openfoodfacts/openfoodfacts-dart/blob/master/lib/model/ProductResult.dart

0 seems to be an error
And 1 is OK

Reply:
Yeah, So thought leaving this as it is would be better